core/domain/acl_decorators.py

```diff
@@ -830,6 +830,28 @@ def test_can_edit(self, topic_id, **kwargs):
     return test_can_edit
 
 
+def can_edit_subtopic_page(handler):
+    """Decorator to check whether the user can edit a subtopic page of a topic.
+    """
+    def test_can_edit(self, topic_id, **kwargs):
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        topic_rights = topic_services.get_topic_rights(topic_id)
+        if topic_rights is None:
+            raise base.UserFacingExceptions.PageNotFoundException
+
+        if topic_services.check_can_edit_topic(self.user, topic_rights):
```

Comment:
check_can_edit_subtopic_page

Split it so that it's an action that can be easily moved to a separate role later if needed. Don't bundle unless that makes sense conceptually (i.e. you cannot imagine doing one without the other).

Reply:
Done.